### PR TITLE
feat: beefy velo vault relayer

### PR DESCRIPTION
**Pull request by piesrtasty**

### CODE REVIEW

File: src/contracts/oracles/BeefyVeloVaultRelayer.sol [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+  /// @inheritdoc IBaseOracle
+  function getResultWithValidity() external view returns (uint256 _result, bool _validity) {
+    (uint256 _token0priceSourceValue, bool _token0priceSourceValidity) = token0priceSource.getResultWithValidity();
+
+    (uint256 _token1priceSourceValue, bool _token1priceSourceValidity) = token1priceSource.getResultWithValidity();
+
+    uint256 _totalValue = _getCombinedPriceValue(_token0priceSourceValue, _token1priceSourceValue);
+
+    _validity = _token0priceSourceValidity && _token1priceSourceValidity;
```
Comment (cupOJoseph) [inline src/contracts/oracles/BeefyVeloVaultRelayer.sol:83]: if 1 is valid but the other is not, you dont want to use the valid one?

File: src/contracts/oracles/BeefyVeloVaultRelayer.sol [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    return _totalValue;
+  }
+
+  /// @notice Returns the combined price of the two tokens in the velo pool
+  function _getCombinedPriceValue(
+    uint256 _token0priceSourceValue,
+    uint256 _token1priceSourceValue
+  ) internal view returns (uint256 _combinedPriceValue) {
+    // 1 mooToken
+    uint256 _mooTokenBalance = 1_000_000_000_000_000_000;
```
Comment (cupOJoseph) [inline src/contracts/oracles/BeefyVeloVaultRelayer.sol:109]: where do you verify this number?

**Review by cupOJoseph — APPROVED** [review]
Logic looks solid to me, but I'm not super familiar with the beefy vault interface here so I'm making some assumptions

File: src/contracts/oracles/BeefyVeloVaultRelayer.sol (hunk 1 above)
Comment (piesrtasty) [reply src/contracts/oracles/BeefyVeloVaultRelayer.sol:83]: The quote for the beefy vault moo token should only be considered valid if the quotes for the 2 underlying tokens are both valid.

File: src/contracts/oracles/BeefyVeloVaultRelayer.sol (hunk 2 above)
Comment (piesrtasty) [reply src/contracts/oracles/BeefyVeloVaultRelayer.sol:109]: I don't follow, it's just the number 1.

File: src/contracts/oracles/BeefyVeloVaultRelayer.sol [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    // # of token0 in velo pool
+    uint256 _reserve0 = veloPool.reserve0();
+    // # of token in velo pool
```
[the remark this answers is not in the document]
Comment (piesrtasty) [reply src/contracts/oracles/BeefyVeloVaultRelayer.sol:113]: good catch! Thank you.

**Pull request merged**